From 7d3070d224323c0d6b596ed2a131fcd04ba450b8 Mon Sep 17 00:00:00 2001
From: Adam Jackson <ajax@nwnk.net>
Date: Thu, 16 Oct 2008 10:24:04 +0800
Subject: [PATCH] Fix Mac mini crash in DDC mode probe

SDVO on Mac mini trys to get EDID from CRT port, which
failed with recent DVI-I change.
---
 src/i830_sdvo.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/src/i830_sdvo.c b/src/i830_sdvo.c
index 8f1b20b..35fccd5 100644
--- a/src/i830_sdvo.c
+++ b/src/i830_sdvo.c
@@ -1495,7 +1495,9 @@ i830_sdvo_get_ddc_modes(xf86OutputPtr output)
     intel_output = crt->driver_private;
     if (intel_output->type == I830_OUTPUT_ANALOG &&
 	crt->funcs->detect(crt) == XF86OutputStatusDisconnected) {
+	I830I2CInit(pScrn, &intel_output->pDDCBus, GPIOA, "CRTDDC_A");
 	edid_mon = xf86OutputGetEDID(crt, intel_output->pDDCBus);
+	xf86DestroyI2CBusRec(intel_output->pDDCBus, TRUE, TRUE);
     }
     if (edid_mon) {
 	xf86OutputSetEDID(output, edid_mon);
-- 
1.6.0.3

